crates/sats/src/algebraic_type.rs

```diff
@@ -121,52 +151,54 @@ impl Default for AlgebraicType {
     }
 }
 
-#[allow(non_upper_case_globals)]
 impl AlgebraicType {
     /// The first type in the typespace.
     pub const ZERO_REF: Self = Self::Ref(AlgebraicTypeRef(0));
 
-    /// The built-in Bool type.
-    pub const Bool: Self = Self::Builtin(BuiltinType::Bool);
-
-    /// The built-in signed 8-bit integer type.
-    pub const I8: Self = Self::Builtin(BuiltinType::I8);
-
-    /// The built-in unsigned 8-bit integer type.
-    pub const U8: Self = Self::Builtin(BuiltinType::U8);
-
-    /// The built-in signed 16-bit integer type.
-    pub const I16: Self = Self::Builtin(BuiltinType::I16);
-
-    /// The built-in unsigned 16-bit integer type.
-    pub const U16: Self = Self::Builtin(BuiltinType::U16);
-
-    /// The built-in signed 32-bit integer type.
-    pub const I32: Self = Self::Builtin(BuiltinType::I32);
-
-    /// The built-in unsigned 32-bit integer type.
-    pub const U32: Self = Self::Builtin(BuiltinType::U32);
+    /// Returns whether this type is the conventional address type.
+    pub fn is_address(&self) -> bool {
+        matches!(self, Self::Product(p) if p.is_address())
+    }
 
-    /// The built-in signed 64-bit integer type.
-    pub const I64: Self = Self::Builtin(BuiltinType::I64);
+    /// Returns whether this type is the conventional identity type.
+    pub fn is_identity(&self) -> bool {
+        matches!(self, Self::Product(p) if p.is_identity())
+    }
 
-    /// The built-in unsigned 64-bit integer type.
-    pub const U64: Self = Self::Builtin(BuiltinType::U64);
+    /// Returns whether this type is scalar or a string type.
+    pub fn is_scalar_or_string(&self) -> bool {
+        self.is_scalar() || self.is_string()
+    }
```

Comment:
This could be considered contentious :)

Is there a specific reason for introducing scalar?